lib/spack/spack/cmd/test.py

```diff
@@ -23,45 +26,67 @@
 
 
 def setup_parser(subparser):
-    subparser.add_argument('--keep-stage', action='store_true',
-                           help='Keep testing directory for debugging')
-    subparser.add_argument(
+    sp = subparser.add_subparsers(metavar='SUBCOMMAND', dest='test_command')
+
+    # Run
+    run_parser = sp.add_parser('run', help=test_run.__doc__)
+
+    run_parser.add_argument(
+        '-n', '--name', help="The name of the test. Default is current time"
+    )
+    run_parser.add_argument('--keep-stage', action='store_true',
```

Comment:
Why overload `--keep-stage` argument to retain testing?  If retain, then perhaps the test help message should reflect its dual use.

I do like having an option for retention BTW.

Reply:
This is something to consider in a separate PR, this option is orthogonal to this PR. It appears in the diff because `spack test run` replaced `spack test` in this PR.

Reply:
That sounds reasonable.